collector/corosync/parser.go

```diff
@@ -211,21 +212,22 @@ func parseMembers(quorumToolOutput []byte) (members []Member, err error) {
 	/*
 	   Membership information
 	   ----------------------
-	      Nodeid      Votes Name
-	   		1          1 192.168.125.24
-	   		2          1 192.168.125.25 (local)
+	   Nodeid      Votes    Qdevice Name
+	        1          1    A,V,NMW nfs01 (local)
+	        2          1    A,V,NMW nfs02
```

Comment:
do these values change at runtime?

Reply:
yes the can change https://susedoc.github.io/doc-sleha/master/single-html/SLE-HA-guide/#co-quorum-nodes

Reply:
hmm, that's what I thought. We shouldn't put them in a label then: I would suggest to not include these values at all at the moment. If we are asked to export them, we'll need to introduce new metrics with static labels.

Reply:
why we shouldn't put them in a label? Can  you elaborate this point why do you think we shouldn't?

Reply:
I there is one key takeaway to be remembered from Promcon, is that most Prometheus users are using labels the wrong way, and we are doing this too, a lot.
Labels are not intended to be used for values that change over the lifecycle of what is meant to be a single timeseries entry.
Prometheus timeseries DB is effectively an indexed key-value store, and labels are part of the key. Every new label value will create a new timeseries.

In this specific case, for example, when the communication between the node and the qdevice changes status, we would end up with two timeseries for the same node.

A metric line exposing nominal status would be like this:
```
ha_cluster_corosync_member_votes{local="false",node="node01",node_id="0",qdevice="A,V,NMW"} 1
```

While a metric exposing a problem would be like this:
```
ha_cluster_corosync_member_votes{local="false",node="node01",node_id="0",qdevice="NA,NV,NMW"} 1
```

The thing that changes is just the `qdevice` label, not **the value of the metric**. These are effectively two different timeseries, they're like two separate entities, instead of being 2 different statuses of the same entity.

What these metrics are meant to represent is just the number of votes, not the status of the communication between the nodes and the qdevice. It doesn't make much sense to count the vote of a node differently when it's in a status rather than another, but this is exactly what happens if we put values that change at runtime in the labels.

This causes _a lot_ of problems when consuming metrics, and in the long term puts a lot of weight in the DB index, eventually causing serious performance problems down the line.

This by far the most common and easiest mistake to make, we are already affected by this in multiple cases, so we need to avoid this in the future, and hopefully fix all the old metrics for v2 of the exporter.

A way to correctly represent the qdevice status of nodes would be a completely new metric dedicated to this, where the status is in a label, but _the label doesn't change over time_, its _value_ does.

```
ha_cluster_corosync_qdevice_status{node="node01",status="NA"} 0
ha_cluster_corosync_qdevice_status{node="node01",status="A"} 0
ha_cluster_corosync_qdevice_status{node="node01",status="V"} 1
ha_cluster_corosync_qdevice_status{node="node01",status="NV"} 0
ha_cluster_corosync_qdevice_status{node="node01",status="NMW"} 1
ha_cluster_corosync_qdevice_status{node="node01",status="MW"} 0
```

This way we'll have a _static_ number of timeseries for a single entity (the node), that _will **not increase** over time_. The only thing changing over time is _their value_.

Reply:
Ok fair point. Regarding the prometheus performance , we don't have any real data and this are just assumptions in the context of our small exporter.

But in sake of the meaning of the metric  make sense.  I will adapt the pr accordingly thx

Reply:
> we don't have any real data and this are just assumptions in the context of our small exporter.

Hmm, just to clarify: these are not my assumptions. Putting dynamic content in labels is a practice abundantly advertised against within the Prometheus community and was a very prominent topic during the last PromCon.

No, I don't have hard data to show, so yes, this is anecdotal, but a lot of people more knowledgeable than me keep saying to not do this, so I think we should listen to them. :wink: 

To put it simply: we have been doing it very wrong all the time; the `handb_exporter`, which puts entire dynamically generated error messages inside labels, is a ticking bomb waiting to explode, and it will cause production workloads to eventually malfunction.

Reply:
I agree with this, but I guess the performance impact is not critical in our exporter . although I agree that we shouldn't overload the label and follow upstream recommendations when they make sense . that is fair point